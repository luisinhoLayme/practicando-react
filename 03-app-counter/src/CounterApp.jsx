import PropTypes from 'prop-types'
import { useState } from 'react'

export const CounterApp = ({ value }) => {
  const [ counter, setCounter ] = useState(value)

  const handleAdd = () => {
    // setCounter(counter + 1)
    setCounter((c) => c + 1) // la c tiene el valor actual del counter
  }

  const handleSubtract = () => {
    setCounter((c) => c - 1)
  }

  const handleReset = () => {
    setCounter(value)
  }

  return (
    <section className="container">
      <h1>Counter App</h1>
      <h3 className="counter">{ counter }</h3>
      <div className="container-buttons">
        <button
          type="button"
          onClick={ handleAdd }
        >
          +1
        </button>
        <button
          type="button"
          onClick={ handleSubtract }
        >
          -1
        </button>
        <button
          type="button"
          onClick={ handleReset }
        >
          reset
        </button>
      </div>
    </section>
  )
}

CounterApp.propTypes = {
  value: PropTypes.number
}

CounterApp.defaultProps = {
  value: 0
}
