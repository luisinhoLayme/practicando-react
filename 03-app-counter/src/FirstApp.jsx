import PropTypes from 'prop-types'

export const FirstApp = ({ title, def }) => {

  const handleSaludo = () => {
    return 'hola como estas lol ☺️:) '
  }


  return (
    <>
      <h1 style={{color: '#f90',}} data-testid="test-title">{ title }</h1>

      <div style={{color: 'limegreen'}}>{ def }</div>
      <div style={{color: 'limegreen'}}>{ def }</div>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  def: PropTypes.number
}

FirstApp.defaultProps = {
  // title: 'el dios ...!',
  def: 444
}
