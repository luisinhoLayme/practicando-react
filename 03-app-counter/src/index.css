
html {
  font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  background-color: #242424;
  display: flex;
  place-content: center;
  min-height: 100vh;
}

.container {
  display: flex;
  flex-direction: column;
  gap: 1rem;
  margin-top: 10rem;
}

h1 {
  color: plum;
  text-align: center;
  margin: 0;
}

.counter {
  color: orange;
  text-align: center;
  margin: 0;
}

.container-buttons {
  display: flex;
  justify-content: center;
  gap: .5rem;
}

.container-buttons button {
  border: none;
  border-radius: 5px;
  color: #fff;
  background-color: #57894C;
  padding: .3rem 1rem;
  transition: background-color .3s ease-in;
}

.container-buttons button:hover {
  background-color: #649D58;
}

.container-buttons button:active {
  scale: .9;
}

