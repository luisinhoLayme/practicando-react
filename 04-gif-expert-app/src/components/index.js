
export * from './Loading'
export * from './Items'
export * from './AddCategory'
export * from './GifGrid'

