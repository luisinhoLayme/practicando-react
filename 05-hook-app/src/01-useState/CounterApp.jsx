import { useState } from 'react'

export const CounterApp = () => {
  const [ counter, setCounter ] = useState({
    counter1: 10,
    counter2: 20,
    counter3: 30
  })

  const { counter1, counter2, counter3 } = counter

  const onCounter = () => {
    setCounter({
      ...counter,
      counter1: counter1 + 1,
    })
  }

  return (
    <div>
      <h2>Counter1: { counter1 }</h2>
      <h2>Counter2: { counter2 }</h2>
      <h2>Counter3: { counter3 }</h2>
      <hr />
      <button onClick={ onCounter }>+1</button>
    </div>
  )
}
