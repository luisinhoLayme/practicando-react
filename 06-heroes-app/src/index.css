:root {
  font-family: Helvetica, Arial, sans-serif;
  line-height: 1.5;
  font-weight: 400;

  color-scheme: light dark;
  color: rgba(255, 255, 255, 0.87);
  background-color: #242424;

  font-synthesis: none;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

a {
  font-weight: 400;
  color: #646cff;
  text-decoration: inherit;
}
a:hover {
  color: #535bf2;
}

a.active {
  color: #B6AFAF;
}

.header {
  background-color: #202020;
  display: grid;
  max-width: 1300px;
  grid-template-columns: repeat(2, 1fr);
  row-gap: .5rem;
  width: 95%;
  margin: .5rem  auto 0 auto;
  position: sticky;
  top: 0;
  z-index: 10;
}

.header::after {
  content: '';
  width: 100%;
  height: .4px;
  background-color: #989898;
  grid-column: 1 / 3;
}

.header__nav {
  display: flex;
  gap: 1rem;
  align-items: center;
}

.header__section {
  display: flex;
  align-items: center;
  justify-content: flex-end;
  gap: .5rem;
}

.header__section span {
  color: #62B7DA;
}

.header__logo {
  color: #8EDE84;
  font-weight: 900;
}

body {
}

main {
  max-width: 1300px;
  width: 95%;
  margin: 1rem auto 0 auto;
}

h1 {
  font-size: 3.2em;
  line-height: 1.1;
}

button {
  border-radius: 8px;
  border: 1px solid transparent;
  padding: 0.6em 1.2em;
  font-size: 1em;
  font-weight: 500;
  font-family: inherit;
  background-color: #1a1a1a;
  cursor: pointer;
  transition: border-color 0.25s;
}
button:hover {
  border-color: #646cff;
}
button:focus,
button:focus-visible {
  outline: 4px auto -webkit-focus-ring-color;
}

@media (prefers-color-scheme: light) {
  :root {
    color: #213547;
    background-color: #ffffff;
  }
  a:hover {
    color: #747bff;
  }
  button {
    background-color: #f9f9f9;
  }
}
