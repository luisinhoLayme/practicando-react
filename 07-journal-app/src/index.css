@tailwind base;
@tailwind components;
@tailwind utilities;


@layer base {
  html {
    font-family: "Lato";
  }
}

.button {
  @apply  text-white rounded-md p-2;
}

.input {
  @apply border-solid outline-none text-sm w-full border-[1px] rounded-md p-2;
}
.input-error {
  @apply w-full border-[1px] border-[#FF6F6F] rounded-md p-2;
}

.gallery::-webkit-scrollbar {
  width: 4px;
}
.gallery::-webkit-scrollbar-track{
  background-color: #D2CDCD;
}
.gallery::-webkit-scrollbar-thumb {
  border-radius: 5px;
  border: 2px solid #686868;
}
