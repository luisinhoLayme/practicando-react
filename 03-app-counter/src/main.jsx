import React from 'react'
import ReactDOM from 'react-dom/client'
// import { AppCounter } from './AppCounter.jsx'
import './index.css'
import { CounterApp } from './CounterApp.jsx'
// import { FirstApp } from './FirstApp'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <CounterApp value={ 5 } />
    {/* <FirstApp title="Hola Luisinho" /> */}
  </React.StrictMode>,
)

//INFO: Hola


