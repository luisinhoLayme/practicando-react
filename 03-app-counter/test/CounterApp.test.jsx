import { render, screen, fireEvent } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe('Pruebas en <CounterApp />', () => {

  const value = 100

  test('Debe de hacer match con el snapshot', () => {

    const { container } = render( <CounterApp value={ value } /> )
    expect( container ).toMatchSnapshot()

  });

  test('Debe de mostrar el valor inicial de 100', () => {

    render( <CounterApp value={ 100 } /> )
    expect( screen.getByText(100) ).toBeTruthy()
    // expect( screen.getByRole('heading', { level: 3 }).innerHTML ).toContain('100')

  })

  test('debe de incrementar con el boton +1', () => {

    render( <CounterApp value={ value } /> )
    fireEvent.click( screen.getByText('+1') )
    expect( screen.getByText('101') ).toBeTruthy()

  })

  test('debe de decrementar con el boton -1', () => {

    render( <CounterApp value={ value } /> )
    fireEvent.click( screen.getByText('-1') )
    expect( screen.getByText('99') ).toBeTruthy()

  })

  test('debe de funcionar el boton de reset', () => {

    render( <CounterApp value={ value } /> )
    fireEvent.click( screen.getByText('+1') )
    fireEvent.click( screen.getByText('+1') )
    fireEvent.click( screen.getByText('+1') )
    // fireEvent.click( screen.getByText('reset') )
    fireEvent.click(screen.getByRole('button', { name: 'btn-reset' }))

    expect( screen.getByText(value) ).toBeTruthy()

  })

})
