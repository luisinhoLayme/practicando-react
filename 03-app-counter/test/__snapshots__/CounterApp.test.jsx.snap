// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> Debe de hacer match con el snapshot 1`] = `
<div>
  <section
    class="container"
  >
    <h1>
      Counter App
    </h1>
    <h3
      class="counter"
    >
      100
    </h3>
    <div
      class="container-buttons"
    >
      <button
        type="button"
      >
        +1
      </button>
      <button
        type="button"
      >
        -1
      </button>
      <button
        aria-label="btn-reset"
        type="button"
      >
        Reset
      </button>
    </div>
  </section>
</div>
`;
